.github/renovate.json5

```diff
@@ -0,0 +1,53 @@
+{
+    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+    "internalChecksFilter": "strict",
+    "labels": ["dependencies"],
+
+    // Wait a few days for any new package, as a precaution against malicious publishes
+    "minimumReleaseAge": "3 days",
+
+    "packageRules": [
+        // Enable major version updates, but don't auto-merge them
+        {
+            "automerge": false,
+            "matchUpdateTypes": ["minor"]
+        },
+
+        // Enable minor version updates with auto-merging
+        {
+            "automerge": true,
+            "automergeStrategy": "squash",
```

Comment:
not yet supported on GitHub .

@rarkins we really should add support for it to GitHub platform 🙃

Reply:
I found this old and closed issue:

- https://github.com/renovatebot/renovate/issues/16814

I think the original posters problem was solved by blocking rebase merges in the GitHub repository settings?

Read the [GitHub Docs, Configuring commit squashing for pull requests](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests) to learn where to find this setting. 😉

Reply:
this option is only supported on bitbucket cloud and gitea

https://docs.renovatebot.com/configuration-options/#automergestrategy